src/PerfTreeParser-Tests/PerfTreeParserTest.class.st

```diff
@@ -468,6 +468,41 @@ PerfTreeParserTest >> testPercentageOf [
 		equals: (testParserSamePercentage percentageOf: 6)
 ]
 
+{ #category : 'tests' }
+PerfTreeParserTest >> testRawPercentageOf [
+	"Cases with 4 digits."
+
+	self assert: (testParserSimple rawPercentageOf: 1) equals: '46.02'.
+
+	self
+		assert: testNodeSimple rawPercentage
+		equals: (testParserSimple rawPercentageOf: 1).
+
+	self
+		assert:
+		(testParserSimple rawPercentageOf:
+			 testParserSimple numberOfFunctions)
+		equals: testNodeSimple traces first rawPercentage.
+
+	"Case with 3 digits."
+	self
+		assert: (testParserMultipleChildren rawPercentageOf: 8)
+		equals: '0.53'.
+
+	self
+		assert: testNodeMultipleChildren traces last rawPercentage
+		equals: (testParserMultipleChildren rawPercentageOf: 8).
+
+	"Case with same percentage"
+	self
+		assert: (testParserSamePercentage rawPercentageOf: 4)
+		equals: (testParserSamePercentage rawPercentageOf: 5).
+
+	self
+		assert: (testParserSamePercentage rawPercentageOf: 5)
+		equals: (testParserSamePercentage rawPercentageOf: 6)
```

Comment:
As a general rule, we should try to not put many unrelated asserts together in the same test.
They are testing different cases, so this means that if the first fails we will not have information about the ones below.

Ideally, they should be split in many tests

Reply:
I split the tests for every big test method that had the same issue